src/main/java/de/zalando/aruha/nakadi/service/subscription/state/StartingState.java

```diff
@@ -0,0 +1,53 @@
+package de.zalando.aruha.nakadi.service.subscription.state;
+
+import de.zalando.aruha.nakadi.exceptions.NoStreamingSlotsAvailable;
+import de.zalando.aruha.nakadi.service.subscription.model.Partition;
+import de.zalando.aruha.nakadi.service.subscription.model.Session;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StartingState extends State {
+    private static final Logger LOG = LoggerFactory.getLogger(StartingState.class);
+
+    @Override
+    public void onEnter() {
+        getZk().runLocked(this::createSubscriptionLocked);
+    }
+
+    /**
+     * 1. Checks, that subscription node is present in zk. If not - creates it.
+     * <p>
+     * 2. Registers session.
+     * <p>
+     * 3. Switches to streaming state.
+     */
+    private void createSubscriptionLocked() {
```

Comment:
This method creates the subscription, registers the session and switches the states. But it's name tells us that it's responsibility is to create the subscription only.